In [ ]:
!pip install mp_api -q

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 97.2/97.2 kB 5.3 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 385.1/385.1 kB 5.5 MB/s eta 0:00:00a 0:00:01
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 118.0/118.0 kB 5.5 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 63.1/63.1 kB 4.7 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 190.5/190.5 kB 6.6 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.0/10.0 MB 5.5 MB/s eta 0:00:0000:010:01
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 561.4/561.4 kB 5.9 MB/s eta 0:00:00a 0:00:01
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 395.2/395.2 kB 4.5 MB/s eta 0:00:00a 0:00:01
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 307.7/307.7 kB 4.7 MB/s eta 0:00:00a 0:00:01
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 138.4/138.4 kB 9.8 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 676.5/676.5 kB 5.5 MB/s eta 0:00:00a 0:00:01
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 5

In [54]:
import pandas as pd
import numpy as np

import sys
sys.path.append('/notebooks/src/')

from Ranker import *


ModuleNotFoundError: No module named 'mp_api'

# Read in all materials and take only flat ones

In [34]:
df = pd.read_csv("csv_data/material_flatness_score_data/Compound_flatness_score_segments.csv")

In [35]:
df_slice = df[['material_id','flat','flat_segments']]
# extract materials with at least one flat band across the whole k-space path chosen
df_flat_only = df_slice.loc[df_slice['Flat/not-flat']==True]
df_flat_only

MP-ID  Flat/not-flat Flat segments
4            mp-7           True           [4]
18          mp-25           True           [3]
26          mp-35           True  [1, 2, 3, 4]
33          mp-43           True  [1, 2, 3, 4]
64          mp-83           True        [3, 4]
...           ...            ...           ...
69787  mp-1443188           True           [4]
69788  mp-1443279           True     [1, 3, 4]
69791  mp-1443442           True        [3, 4]
69806  mp-1445771           True  [1, 2, 3, 4]
69816  mp-1475733           True     [1, 2, 4]

[33399 rows x 3 columns]

# Add synthNN scores

In [36]:
df_synthNN_scores = pd.read_csv("csv_data/SynthNN_final_scores.csv")
df_synthNN_scores

mp-ID       Formula  SynthNN_score
0            mp-7             S       0.013924
1           mp-25            N2       0.127548
2           mp-35            Mn       0.007099
3           mp-43             U       0.001247
4           mp-83             S       0.013924
...           ...           ...            ...
33394  mp-1443188        ZnCuF4       0.574649
33395  mp-1443279       Cr(OF)3       0.035679
33396  mp-1443442           VF5       0.727455
33397  mp-1445771      K2P2H2O7       0.974588
33398  mp-1475733  PdAu2(NCl2)4       0.010389

[33399 rows x 3 columns]

In [37]:
df_flat_only['Formula']= df_synthNN_scores['Formula'].values
df_flat_only['SynthNN_score']= df_synthNN_scores['SynthNN_score'].values
df_flat_only

/tmp/ipykernel_549/1011652856.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_flat_only['Formula']= df_synthNN_scores['Formula'].values
/tmp/ipykernel_549/1011652856.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_flat_only['SynthNN_score']= df_synthNN_scores['SynthNN_score'].values


MP-ID  Flat/not-flat Flat segments       Formula  SynthNN_score
4            mp-7           True           [4]             S       0.013924
18          mp-25           True           [3]            N2       0.127548
26          mp-35           True  [1, 2, 3, 4]            Mn       0.007099
33          mp-43           True  [1, 2, 3, 4]             U       0.001247
64          mp-83           True        [3, 4]             S       0.013924
...           ...            ...           ...           ...            ...
69787  mp-1443188           True           [4]        ZnCuF4       0.574649
69788  mp-1443279           True     [1, 3, 4]       Cr(OF)3       0.035679
69791  mp-1443442           True        [3, 4]           VF5       0.727455
69806  mp-1445771           True  [1, 2, 3, 4]      K2P2H2O7       0.974588
69816  mp-1475733           True     [1, 2, 4]  PdAu2(NCl2)4       0.010389

[33399 rows x 5 columns]

# Add sublattice elements 
### **Note at this point, we cut out rows from the 33399 array to match the smaller 33251 array which was actually clustered
### When adding e.g. CNN_synth scores, we will need to add it above, before this cut is made, so that all rows match up

In [38]:
df_sublattices = pd.read_csv("csv_data/material_flatness_score_data/Flat_band_sublattices.csv")
df_sublattices.rename(columns={'MP_index': 'MP-ID'}, inplace=True)
df_sublattices.drop(columns=['Unnamed: 0'], inplace=True)

In [39]:
merged_df = pd.merge(df_flat_only, df_sublattices, on='MP-ID', how='inner')
merged_df

MP-ID  Flat/not-flat Flat segments       Formula  SynthNN_score  \
0            mp-7           True           [4]             S       0.013924   
1           mp-25           True           [3]            N2       0.127548   
2           mp-35           True  [1, 2, 3, 4]            Mn       0.007099   
3           mp-43           True  [1, 2, 3, 4]             U       0.001247   
4           mp-83           True        [3, 4]             S       0.013924   
...           ...            ...           ...           ...            ...   
33246  mp-1443188           True           [4]        ZnCuF4       0.574649   
33247  mp-1443279           True     [1, 3, 4]       Cr(OF)3       0.035679   
33248  mp-1443442           True        [3, 4]           VF5       0.727455   
33249  mp-1445771           True  [1, 2, 3, 4]      K2P2H2O7       0.974588   
33250  mp-1475733           True     [1, 2, 4]  PdAu2(NCl2)4       0.010389   

       Cluster_ID  cluster_size sublattice_element  dbcv_cluster  \
0              -1         24875                  S      0.846820   
1              63           282                  N      0.212178   
2              -1         24875                 Mn      0.846820   
3              -1         24875                  U      0.846820   
4              38            22                  S      0.468464   
...           ...           ...                ...           ...   
33246          -1         24875                 Cu      0.846820   
33247          -1         24875                  F      0.846820   
33248          -1         24875                  F      0.846820   
33249          -1         24875                  O      0.846820   
33250          -1         24875                  N      0.846820   

       probabilities  
0           0.000000  
1           0.104165  
2           0.000000  
3           0.000000  
4           0.972129  
...              ...  
33246       0.000000  
33247       0.000000  
33248       0.000000  
33249       0.000000  
33250       0.000000  

[33251 rows x 10 columns]

# Add materials project information